src/focus.service.ts

```diff
@@ -234,6 +234,7 @@ function isNodeAttached(node: HTMLElement, root: HTMLElement) {
 
 @Injectable()
 export class FocusService {
+  public enableRayTrace = true;
```

Comment:
I would discourage you from following that path. Try to keep state for DOM-related behavior in the DOM.